### ISSUE TITLE
A Collisionshape2D Shape which is a line and can take Poolvector2array as points.

### ISSUE THREAD

**Issue by TanosOP** [godotengine/godot-proposals#1587]
**Describe the project you are working on:**
I am working on a game in which user can draw line using Line2D. I have a RigidBody2D(it's mode is set as kinematic) which has Line2D and CollisionShape2D as children. I use get_global_mouse_position() to add points to the Line2D. When the user releases mouse I want it to generate a LineShape for the CollisionShape2D node and set the RigidBody2D mode to Rigid
**Describe the problem or limitation you are having in your project:**
I can't seem to properly use a CollisionShape generated using SegementShape2D's. And there isn't any other collision shape which acts like a normal line.
**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
It will help to overcome the problem as I could use the Line2D points to generate a LineShape2D for the CollisionShape2D node which is precisely what I need.
**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
if Input.is_action_just_released("click"):
    var lineshape = LineShape2D.new()
    lineshape.points = get_node("Line2D").points
    get_node("CollisionShape2D").shape = lineshape
**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
For my case, It can't be worked around as I tried using SegementShape2D's but that didn't work too.
**Is there a reason why this should be core and not an add-on in the asset library?:**
I think LineShape2D can be used in many Projects by various people and will be a huge help for me too.

**Comment by madmiraal** [comment]
You could iterate through your points and use the [`CollisionObject2D`](https://docs.godotengine.org/en/stable/classes/class_collisionobject2d.html) API to add multiple `Line2D` shapes to your `CollisionObject2D`.

**Comment by ThanosOP** [comment]
I see what you mean by that but it causes weird behaviour once the body's mode is set to RIGID:
![ezgif com-gif-maker](https://user-images.githubusercontent.com/59230561/94984015-5886d800-0565-11eb-8655-f3fb53108ac3.gif)

**Comment by CaptainSusuwatari** [comment]
Yes, but the resulting collision object (SegmentShape2D) is still missing the width of a drawn Line2D.

**Comment by madmiraal** [comment]
Then you're not creating a line, you're creating a polygon. However the same principal about creating the `CollisoinShape2D` still applies. You can either use two `SegmentShape2D`s (with end caps) or a series of `RectangleShape2D`s.

**Comment by timcode77** [comment]
I suggest that 

> Yes, but the resulting collision object (SegmentShape2D) is still missing the width of a drawn Line2D.

Additionally, if a Line2D has a curve defining its width, the associated collision shape would not have this either.
